### PR TITLE
MBS-7323: Add option to show activity names

### PR DESCRIPTION
**Pull request by sh-csg**

### CODE REVIEW

**Comment by PhMemmel** [comment]
At first sight, basic functionality looks good. More detailed review will follow.

What I noticed from first tests: When dragging the labels around in the teacher's edit mode there is often some "jumping" of the label which is quite irritating. Is it possible to keep the relative position of the label to the mouse pointer after the click while dragging?

**Comment by sh-csg** [comment]
Should be fixed now!

**Comment by PhMemmel** [comment]
> Should be fixed now!

Yes, it is, good job!

**Review by PhMemmel — COMMENTED** [review]
Review round 1 finished. It works like expected and is a nice addition, well done! A few thoughts:

- Code is becoming more and more complex, so the need for automatic tests as well as a refactoring of the data structures increases (discussed internally, how to adress this).
- As there is a new placestore attribute being added, do we need the usual "placestore update workflow" including a version bump to ensure older versions do include it as well?
- For long activity names the handling is becoming a bit difficult. You might want to consider line breaks or ellipsis for the text label?
- Activity names are not being updated in edit mode. If you add a learningmap and enable the display of activity names and save, then rename one of the activities in the learningmap, it will update the view of the learningmap. But if you edit the learningmap, the old activity name will be shown until you save it again which is confusing for the teacher.
- Moving the texts around in edit mode is not possible on mobile devices. Not sure, if this would be easy to add?
- Could you please rebase the branch to latest master?

Thanks for the work and good job!

File: amd/src/learningmap.js [hunk 1]
```diff
@@ -558,16 +630,20 @@ export const init = () => {
...
         let link = document.createElementNS('http://www.w3.org/2000/svg', 'a');
         link.setAttribute('id', id);
         link.setAttribute('xlink:href', '');
         link.appendChild(child);
         if (!(title === null)) {
             link.appendChild(title);
         }
+        if (!(text === null)) {
+            link.appendChild(text);
+        }
```
[the remark this answers is not in the document]
Comment (sh-csg) [reply amd/src/learningmap.js:646]: No special reason

**Comment by sh-csg** [comment]
Everything should be fixed now, except for the wrapping of long activity names as SVG has no wrapping capability itself.
Moving the text on mobile devices works here.

File: amd/src/learningmap.js [hunk 2]
```diff
@@ -515,6 +568,25 @@ export const init = () => {
...
+     * @param {*} x x coordinate of the text
+     * @param {*} y y coordinate of the text
+     * @returns {any}
+     */
+     function text(id, content, x, y) {
+        let text = document.createElementNS('http://www.w3.org/2000/svg', 'text');
+        text.setAttribute('id', id);
+        text.setAttribute('x', x);
+        text.setAttribute('y', y);
+        text.setAttribute('dx', 10 * 1.5);
```
[the remark this answers is not in the document]
Comment (PhMemmel) [reply amd/src/learningmap.js:483]: maybe the introduction of a constant for the circle radius would be helpful, so we have no magic numbers here

File: amd/src/learningmap.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (PhMemmel) [reply amd/src/learningmap.js:545]: want to change it? I think it's a bit easier to read?

**Review by PhMemmel — COMMENTED** [review]
Review round 2 finished, looks good, all the things mentioned have been resolved, just some additional comments

**Comment by PhMemmel** [comment]
Also the refactoring to handle the advanced options is a nice addition, gj!

File: amd/src/learningmap.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (PhMemmel) [reply amd/src/learningmap.js:547]: OK, we can leave it, it also appears on some other places in the code :)

**Comment by PhMemmel** [comment]
Final manual tests finished, no regressions could be found so far.

Nice new feature while keeping the code as maintainable as possible, this should be good to go live.

**Pull request merged**